src/ser/mod.rs

```diff
@@ -61,38 +67,69 @@ impl StdError for Error {
     }
 }
 
+/// Pretty serializer state
 struct Pretty {
     indent: usize,
 }
 
+/// Pretty serializer configuration
+pub struct PrettyConfig {
+    /// New line string
+    pub new_line: String,
+    /// Indentation string
+    pub indentor: String,
```

Comment:
Could theoretically use `&str` or `&[u8]` instead.

Reply:
I tried, facing the lifetime hell since the `&'a` (assuming we don't want `&'static`) starts polluting `Serializer` struct

Reply:
Yeah, let's leave it like that then. Pretty serialization isn't something that suffers from two string allocations I guess.